src/select.cpp

```diff
@@ -99,22 +99,19 @@ zmq::select_t::find_fd_entry_by_handle (fd_entries_t &fd_entries,
          ++fd_entry_it)
         if (fd_entry_it->fd == handle_)
             break;
-    zmq_assert (fd_entry_it != fd_entries.end ());
 
     return fd_entry_it;
 }
 
-void zmq::select_t::rm_fd (handle_t handle_)
+bool zmq::select_t::try_remove_fd_entry (
```

Comment:
This will trigger static analysers and picky compiler since it's not used elsewhere, so it's probably a good idea to wrap it in \#ifdef windows